### ISSUE TITLE
False negative: missing required kw_only argument from parent dataclass not detected

### ISSUE THREAD

**Issue by gitraffe** [astral-sh/ty#3133]
### Summary

When a parent dataclass uses `kw_only=True` and defines a required field (no default), child dataclass constructions that omit that field are not flagged.

```py
from dataclasses import dataclass

@dataclass(kw_only=True, frozen=True)
class Base:
    y: int

@dataclass(frozen=True)
class Child(Base):
    z: str

Child(z="hi")  # should error: missing required keyword argument 'y'
```

At runtime this raises `TypeError: Child.__init__() missing 1 required keyword-only argument: 'y'`.

pyright correctly reports: `Argument missing for parameter "y" (reportCallIssue)`

ty reports: `All checks passed!`

### Version

ty 0.0.25 (d60899a14 2026-03-24)

**Comment by carljm** [comment]
I can't reproduce this. On the [playground](https://play.ty.dev/aa33250a-03bc-440c-aac3-b65c20ae4d61) (which runs ty main branch) this emits the expected error. And I also see that same error on ty 0.0.23, 0.0.24, and 0.0.25.

**Comment by carljm** [comment]
Closing since I can't reproduce, but happy to reopen given a working reproduction!

**Comment by gitraffe** [comment]
My apologies, I had a config problem in my `pyproject.toml`—problem in chair, not in computer. Thank you for the fast response!

**Comment by carljm** [comment]
No problem, thanks for reporting issues!
